packages/editor/src/math/editor.tsx

```diff
@@ -157,10 +159,12 @@ export function MathEditor(props: MathEditorProps) {
   }
 
   function renderControlsPortal(children: JSX.Element) {
-    return createPortal(
-      children,
-      document.querySelector<HTMLDivElement>('.toolbar-controls-target') ??
-        document.body
-    )
+    const root = containerRef.current?.getRootNode()
+    const target =
+      (root instanceof ShadowRoot || root instanceof Document
```

Comment:
nit:

Would be nice to define most booleans here as consts for readability:

e.g:

```
const isShadowDom = root instanceof ShadowRoot

```